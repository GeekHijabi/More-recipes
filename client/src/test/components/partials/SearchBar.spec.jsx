import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ConnectedSearchBar, { SearchBar }
  from '../../../js/components/Partials/SearchBar';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);


const props = {
  apiSearchRecipe: jest.fn(() => Promise.resolve({ ok: true })),
  searchItem: jest.fn(() => Promise.resolve({ ok: true }))
};
describe('Component: SearchBar', () => {
  beforeEach(() => {
    global.toastr = {
      success: () => {},
      error: () => {}
    };
  });

  describe('SearchBar in snapshot', () => {
    it('should render correctly', () => {
      const wrapper = shallow(<SearchBar {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('onChange', () => {
    it('should change event', () => {
      const event = {
        target: { name: 'SearchResults', value: 'rice' },
      };
      const wrapper = shallow(<SearchBar {...props} />);
      wrapper.instance().onChange(event);
      expect(wrapper.instance().state.SearchResults).toEqual('rice');
      expect(wrapper.instance().props.apiSearchRecipe).toBeCalled();
    });
  });

  describe('Connected SearchBar component', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        recipe: {
          searchResult: []
        }
      });
      const wrapper = shallow(<ConnectedSearchBar store={store} />);
      expect(wrapper.length).toBe(1);
    });
  });
});

