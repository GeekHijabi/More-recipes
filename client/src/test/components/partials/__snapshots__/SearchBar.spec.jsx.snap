// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: SearchBar SearchBar in snapshot should render correctly 1`] = `
<div
  className="search"
>
  <div
    className="form-inline"
  >
    <input
      aria-label="Search"
      className="form-control mr-sm-2"
      id="search-bar"
      name="SearchResults"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </div>
</div>
`;
